### PR TITLE
add KubernetesClient.Aot to support Aot 

### PR DESCRIPTION
**Pull request by tg123**
#1395

the main project cannot support aot due to dependencies or some dynamic requirements, for example crd json/yaml serialization
the solution now is doing the same as `.Classic`, to create a separate project, trim aot non friendly features

Known unsupported features
- Watch
- LeaderElection (watch related)
- Customized Patch Object

@brendanburns please also register `KubernetesClient.Aot` to nuget.org

### CODE REVIEW

**Comment by guillaume-chervet** [comment]
@tg123  I hope you will succeed :)

I'am waiting a lot for this awesome PullRequest :)

**Comment by tg123** [comment]
i am also working on making watch for aot
need some big refactor and will try to merge 2 projects together

**Comment by brendandburns** [comment]
I need a `.nuget` file to add it to nuget.org. Shall I wait until this merges?

**Comment by tg123** [comment]
lets merge first and get nuget from github action

**Comment by guillaume-chervet** [comment]
@tg123 it would be awesome :)

File: src/KubernetesClient.Aot/KubernetesClient.Aot.csproj [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    <Compile Include="..\KubernetesClient\Authentication\TokenFileAuth.cs" />
+    <Compile Include="..\KubernetesClient\Authentication\BasicAuthenticationCredentials.cs" />
+    <Compile Include="..\KubernetesClient\Authentication\ITokenProvider.cs" />
+    <Compile Include="..\KubernetesClient\Authentication\ServiceClientCredentials.cs" />
+    <Compile Include="..\KubernetesClient\Authentication\StringTokenProvider.cs" />
+    <Compile Include="..\KubernetesClient\Authentication\TokenCredentials.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="System.Net.Http" Condition="'$(TargetFramework)' == 'net48'" />
```
Comment (WeihanLi) [inline src/KubernetesClient.Aot/KubernetesClient.Aot.csproj:104]: Seemed this can be removed since we target for `net8.0` only

Comment (tg123) [reply src/KubernetesClient.Aot/KubernetesClient.Aot.csproj:104]: thanks, copy paste issue, will fix it

**Comment by guillaume-chervet** [comment]
Thank you @tg123 will test it today :)

**Comment by tg123** [comment]
@guillaume-chervet thank you

**Pull request merged**